packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -65,19 +65,49 @@ def processable_bet(self, bet: Bet) -> bool:
         within_safe_range = now < bet.openingTimestamp + self.params.safe_voting_range
         within_ranges = within_opening_range and within_safe_range
 
-        # if we should not rebet, we have all the information we need
-        if not self.should_rebet:
-            return within_ranges
-
-        # if we should rebet, we should have at least one bet processed in the past
-        if not bool(bet.n_bets):
-            return False
-
         # create a filter based on whether we can rebet or not
         lifetime = bet.openingTimestamp - now
         t_rebetting = (lifetime // UNIX_WEEK) + UNIX_DAY
         can_rebet = now >= bet.processed_timestamp + t_rebetting
-        return within_ranges and can_rebet
+        bet_id = bet.id
+
+        # if we should not rebet we need to ignore markets that have been bet upon in the past
+        if not self.should_rebet:
+            if bet.n_bets != 0:
+                return False
+            else:
+                # filter for changed liquidity on previously processed markets to avoid resampling the same market repeatedly
+                if bet.processed_timestamp > 0:
+                    scaled_liquidity_changed = (
+                        bet.scaledLiquidityMeasure
+                        != self.shared_state.bet_selection_stats.get(bet_id, {}).get(
+                            "scaledLiquidityMeasure", 0
+                        )
+                    )
+                    return within_ranges and can_rebet and scaled_liquidity_changed
+                else:
+                    return within_ranges and can_rebet
```

Comment:
Why check for `can_rebet` since we are inside the `if not self.should_rebet:` condition?

Reply:
As Juli and I had understood, can_rebet checks that once a bet has been sampled and either been bet upon or deemed unprofitable , enough time has passed that it can be resampled.